File: backend/internal/dto/boundary_proposal.go
package dto

type ProposalQuery struct {
	ParcelID *uint
	State    string
	Page     int
	PageSize int
}

type CreateProposalRequest struct {
	ParcelID        uint    `json:"parcel_id" validate:"required,gt=0"`
	BaseVersion     uint    `json:"base_version" validate:"required,gt=0"`
	ProposedGeoJSON string  `json:"proposed_geojson" validate:"required"`
	ObservationIDs  []uint  `json:"observation_ids" validate:"max=200"`
	SnapToleranceM  float64 `json:"snap_tolerance_m" validate:"required,gt=0,lte=1000"`
	Rationale       string  `json:"rationale" validate:"max=2000"`
}

type ProposalTransitionRequest struct {
	To        string `json:"to" validate:"required"`
	Version   uint   `json:"version" validate:"required,gt=0"`
	Rationale string `json:"rationale" validate:"max=2000"`
}

File: frontend/src/api/boundary-proposal.ts
import { api } from './client'
import type { ApiEnvelope } from '@/types/api'
import type { BoundaryProposal } from '@/types/boundary-proposal'
import type { ProposalState } from '@/types/enums/proposal-state'

export interface ProposalQuery {
  parcel_id?: number
  state?: ProposalState
  page?: number
  page_size?: number
}

export interface ProposalCreateInput {
  parcel_id: number
  base_version: number
  proposed_geojson: string
  observation_ids: number[]
  snap_tolerance_m: number
  rationale: string
}

export interface ProposalTransitionInput {
  to: ProposalState
  version: number
  rationale?: string
}

export const boundaryProposalApi = {
  list: (params?: ProposalQuery) => api.get<ApiEnvelope<BoundaryProposal[]>>('/proposals', { params }),
  detail: (id: number) => api.get<ApiEnvelope<BoundaryProposal>>(`/proposals/${id}`),
  create: (body: ProposalCreateInput) => api.post<ApiEnvelope<BoundaryProposal>>('/proposals', body),
  transition: (id: number, body: ProposalTransitionInput) =>
    api.post<ApiEnvelope<BoundaryProposal>>(`/proposals/${id}/transition`, body),
}
